### ISSUE TITLE
[V3] Implement new Concept Exercise: Function Expression and ES6 Arrow Function 

### ISSUE THREAD

**Issue by shubhsk88** [exercism/javascript#955]
This issue describes how to declare the function expression and declaration with the ES6 Arrow function. 

This issue describes how to implement the `function`  concept exercise for the Javascript track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

This exercise aims to tell the student about the declaration of `function,` `arrow function.`

For example:
- `function hello ( ) { }`
- `let hello = function ( ) { }`
- `const hello = ( ) => { }`

## Learning objectives

- The student able to figure out the difference between function declaration and function expression.
- The student able to know how the arrow function can be implemented and where to use it.
-  Explaining the definition of high order function 

## Out of scope
- Closures
- Use of arrow function in the context of `this` keyword;
- Callbacks

##  Concepts

- `function` declartions

## Prerequisites
-  `functions-intro`
- `variables`

### After

- Understanding of execution context of the function 
- Higher-order function
- When to use function declarations or function expressions 

## Help

If you have any questions while implementing the exercise, please post the questions as comments on this issue.

**Comment by junedev** [comment]
I am closing this issue because we decided to cut the concepts/concept exercises differently. See https://github.com/exercism/javascript/issues/951#issuecomment-778834953 for details. Arrow functions will be combined with callbacks in one exercise and function expression are covered in the functions concept/concept exercise.
